sentry_sdk/tracing.py

```diff
@@ -435,6 +457,15 @@ def __init__(
                 "instead of Span(transaction=...)."
             )
             name = kwargs.pop("transaction")
+
+        if "_span_id_generator" in kwargs:
+            raise TypeError("Argument _span_id_generator can only be provided to Spans")
```

Comment:
Transactions also have a `span_id`, right? They have `event_id`, `trace_id` and `span_id`.